### PR TITLE
solution -- pay to pin

### PR DESCRIPTION
**Pull request by antdke**
Hi @ryanckulp, this is my submission for Issue #4 

Here's what I did, high level:

1. On frontend, added text + buy link for users to pay to pin their tweet
2. Created a Stripe Payment Link for a "Pay to Pin product" and added that to the buy link's `href` tag
3. (Hard part) Implemented ActiveMailer and configured it with my Gmail in order to send emails from the backend
4. Created a "Thank you" page where users will redirect to after a successful payment
5. Made it so that once on the "Thank you" page, ActiveMailer sends me an email notifying me of the successful payment with the tweet_url  of the tweet that needs to be pinned.

Lmk your thoughts!

(I probably could've been better with comments)

### CODE REVIEW

**Comment by ryanckulp** [comment]
awesome work @antdke ! this one was a beast.

we need to make 1 change, and it's a big one, but this will really tighten things up.

* as you probably realized, using `Feedback.last` in our mailer is risky; if 2 people create a tweet at a similar time we'll Pin the wrong Tweet
* thus, the mailer function needs `tweet_url` to be passed in as an argument
* in order to trigger this mailer with the correct tweet, the invocation from `pages#thanks` needs access to it
* you can pass the Tweet's ID (primary key `feedback.id`, not `tweet_id`) by setting `session[:feedback_id]` in the same place that `session[:tweet_url]` is set
* finally, use this `feedback_id` variable from your session to produce a payment link with dynamic redirect params using [these docs](https://stripe.com/docs/payments/payment-links/api#tracking-payments)

TLDR - after paying for a Pin, user should be redirected to `/thanks?feedback_id=X`, where `X` is then passed to the mailer.

and bonus points -- user should still land on just `/thanks` -- the `feedback_id` param should be destroyed at the `pages#thanks` controller level after use. if not, a user can refresh `/thanks?feedback_id=X` over and over again, destroying your inbox and causing confusion to boot. a hint on this: redirect purchase somewhere like `/payment_received?feedback_id=X`, and in that* action handle the mailer + redirect to plain `/thanks` view.

good luck :)

**Comment by antdke** [comment]
@ryanckulp done!

What I did:

1. Created a session variable to holds the user's feedback’s id
2. Implemented Stripe's API
3. Used Stripe's API to create a dynamic redirect link (with feedback_id as a parameter) from the Payment Link
4. Created stripe_service.rb to run the Payment Link code—input: feedback_id, output: fresh payment link on each execution
5. Buy link on the frontend changes dynamically with each feedback submission
6. Got ActionMailer to grab the feedback_id from `params`
7. After purchase, user is redirected to payment_received page with the feedback_id params, then instantly redirected to a plain thanks page (no params)

Lmk your thoughts!

(Btw, wasn't sure what you meant by "the feedback_id param should be destroyed". Does what I did count as "destroying it"?)

**Comment by ryanckulp** [comment]
@antdke good updates! i'm going to merge, then PR some improvements with notes for you. will also email you to arrange bounty payment.
